File: app/src/main/java/br/com/ufpe/cin/myfootprints/FriendsFragment.java
package br.com.ufpe.cin.myfootprints;

import android.app.Activity;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.app.ListFragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.TextView;

import com.google.android.gms.flags.impl.DataUtils;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class FriendsFragment extends ListFragment {

    private FriendSharedLocationDAO dbInstance;
    private  String[] arrayAdapterData;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup parent, Bundle savedInstanceState) {
        View v =  inflater.inflate(R.layout.friends_view, parent, false);
        dbInstance = FriendSharedLocationDAO.getInstance(parent.getContext());

        List<String> friends = dbInstance.getFriendSharedLocations();
        FriendSharedLocationParser parser = new FriendSharedLocationParser();

        ArrayList<String>  adapterData = new ArrayList<String>();
        for(String friend: friends){
            boolean hasLocationData = parser.locationUpdatesFromSMSText(friend).getPath().size() > 0;
            boolean isDifferentFromPrevious = adapterData.isEmpty() || !friend.equals(adapterData.get(adapterData.size()-1));
            if (hasLocationData && isDifferentFromPrevious)
                adapterData.add(friend);
        }

        arrayAdapterData = new String[adapterData.size()];
        arrayAdapterData = adapterData.toArray(arrayAdapterData);

        FriendAdapter adapter= new FriendAdapter(getActivity(),arrayAdapterData);
        // Bind adapter to the ListFragment
        setListAdapter(adapter);
        //  Retain the ListFragment instance across Activity re-creation
        setRetainInstance(true);
        setListAdapter(adapter);
        return v;
    }

    @Override
    public void onListItemClick(ListView l, View v, int position, long id) {
        FragmentTransaction ft = getFragmentManager().beginTransaction();
        FriendSharedLocationParser parser = new FriendSharedLocationParser();

        MapFragment f = MapFragment.newInstance(parser.locationUpdatesFromSMSText(arrayAdapterData[position]));
        f.show(ft, "test");

    }


}

File: app/src/main/java/br/com/ufpe/cin/myfootprints/FriendsFragment.kt
package br.com.ufpe.cin.myfootprints

import android.app.Activity
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentTransaction
import android.support.v4.app.ListFragment
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.EditText
import android.widget.ListView
import android.widget.TextView

import com.google.android.gms.flags.impl.DataUtils

import java.util.ArrayList
import java.util.Arrays

class FriendsFragment : ListFragment() {

    private var dbInstance: FriendSharedLocationDAO? = null
    private var arrayAdapterData: Array<String>? = null

    override fun onCreateView(inflater: LayoutInflater, parent: ViewGroup?, savedInstanceState: Bundle?): View? {
        val v = inflater.inflate(R.layout.friends_view, parent, false)
        dbInstance = FriendSharedLocationDAO.getInstance(parent!!.context)

        val friends = dbInstance!!.friendSharedLocations
        val parser = FriendSharedLocationParser

        val adapterData = ArrayList<String>()
        for (friend in friends) {
            val hasLocationData = parser.locationUpdatesFromSMSText(friend).path!!.size > 0
            val isDifferentFromPrevious = adapterData.isEmpty() || friend != adapterData[adapterData.size - 1]
            if (hasLocationData && isDifferentFromPrevious)
                adapterData.add(friend)
        }


        val arrayData = arrayOfNulls<String>(adapterData.size)
        arrayAdapterData = adapterData.toArray(arrayData)
        val data = arrayAdapterData

        val adapter = FriendAdapter(activity, data)
        // Bind adapter to the ListFragment
        listAdapter = adapter
        //  Retain the ListFragment instance across Activity re-creation
        retainInstance = true
        listAdapter = adapter
        return v
    }

    override fun onListItemClick(l: ListView?, v: View?, position: Int, id: Long) {
        val ft = fragmentManager!!.beginTransaction()
        val parser = FriendSharedLocationParser

        val f = MapFragment.newInstance(parser.locationUpdatesFromSMSText(arrayAdapterData!![position]))
        f.show(ft, "test")

    }


}
